main.go

```diff
@@ -133,6 +137,46 @@ func (d *Downloader) downloadFileWithTimeout(userCtx context.Context, u string)
 	}
 }
 
+func (d *Downloader) getSize(ctx context.Context, u string) (uint64, error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodHead, u, nil)
+	if err != nil {
+		return 0, fmt.Errorf("error creating the request for %s: %w", u, err)
+	}
+	resp, err := d.client.Do(req)
+	if err != nil {
+		return 0, fmt.Errorf("error sending a get http request to %s: %w", u, err)
+	}
+	if resp.StatusCode != 200 {
+		return 0, fmt.Errorf("status code is not valid: %v", resp.StatusCode)
```

Comment:
```suggestion
		return 0, fmt.Errorf("error fetching content size -- status code is not valid: %v", resp.StatusCode)
```

Reply:
I think I disagree with this suggested change. The _error fetching content size_ message should be wrapped by the caller of `getSize` in case `getSize` returns an error; not within `getSize`'s implementation (same for the similar suggested change below). Does that make sense?